platform/android/MapLibreAuto/src/main/java/org/maplibre/android/maps/TransformAuto.java

```diff
@@ -0,0 +1,197 @@
+package org.maplibre.android.maps;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+
+import org.maplibre.android.camera.CameraPosition;
+import org.maplibre.android.camera.CameraUpdate;
+import org.maplibre.android.camera.CameraUpdateFactory;
+
+public final class TransformAuto extends Transform {
+
+    private static final String TAG = "GrabMap-Auto-Transform";
+
+    private final NativeMap nativeMap;
+    @Nullable
+    private MapView mapView;
+    @Nullable
+    private MapSurface mapSurface;
+
+    private final Handler handler = new Handler(Looper.getMainLooper());
+
+    @Nullable
+    private CameraPosition cameraPosition;
+    @Nullable
+    private MapLibreMap.CancelableCallback cameraCancelableCallback;
+    private final CameraChangeDispatcher cameraChangeDispatcher;
+
+    private final MapView.OnCameraDidChangeListener moveByChangeListener = new MapView.OnCameraDidChangeListener() {
+        @Override
+        public void onCameraDidChange(boolean animated) {
+            if (animated) {
+                cameraChangeDispatcher.onCameraIdle();
+                removeOnCameraDidChangeListener(this);
+            }
+        }
+    };
+
+    TransformAuto(@NonNull MapView mapView, NativeMap nativeMap, CameraChangeDispatcher cameraChangeDispatcher) {
+        super(mapView, nativeMap, cameraChangeDispatcher);
+        this.mapView = mapView;
+        this.nativeMap = nativeMap;
+        this.cameraChangeDispatcher = cameraChangeDispatcher;
+    }
+
+    TransformAuto(@NonNull MapSurface mapSurface, NativeMap nativeMap, CameraChangeDispatcher cameraChangeDispatcher) {
+        super(null, nativeMap, cameraChangeDispatcher);
+        this.mapSurface = mapSurface;
+        this.nativeMap = nativeMap;
+        this.cameraChangeDispatcher = cameraChangeDispatcher;
+    }
+
+    void initialise(@NonNull MapLibreMap maplibreMap, @NonNull MapLibreMapOptions options) {
+        CameraPosition position = options.getCamera();
+        if (position != null && !position.equals(CameraPosition.DEFAULT)) {
+            moveCamera(maplibreMap, CameraUpdateFactory.newCameraPosition(position), null);
+        }
+        setMinZoom(options.getMinZoomPreference());
+        setMaxZoom(options.getMaxZoomPreference());
+        setMinPitch(options.getMinPitchPreference());
+        setMaxPitch(options.getMaxPitchPreference());
+    }
+
+    //
+    // Camera API
+    //
+
+    @Nullable
+    @UiThread
+    public CameraPosition getCameraPosition() {
+        if (cameraPosition == null) {
+            cameraPosition = invalidateCameraPosition();
+        }
+        return cameraPosition;
+    }
```

Comment:
from testing this pr, this method causes the camera position to _always_ be 0,0, even when the underlying native map has the correct position. i believe that this is because `cameraPosition` is never set here beyond the first `invalidateCameraPosition`, and the parent `Transform` class has its own `cameraPosition`.

what if we delegate this method to the parent and remove the `cameraPosition` variable from this class altogether? i tested this and it fixed the camera position for me.